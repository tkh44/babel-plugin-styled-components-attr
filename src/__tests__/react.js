/* eslint-disable jsx-quotes,no-useless-escape,no-template-curly-in-string */
/* eslint-env jest */
import 'jest-styled-components'
import React from 'react'
import renderer from 'react-test-renderer'
import styled, { css } from 'styled-components'

describe('react', () => {
  test('attr', () => {
    const baseText = css`color: attr(color, #a9e34b);`

    const H1 = styled.h1`
      font-size: attr(fontSize);
      margin: attr(margin rem, 4);
      ${baseText}
    `

    const Title = ({ title }) => {
      return (
        <H1 color="#495057" fontSize={48}>
          {title}
        </H1>
      )
    }

    const tree = renderer.create(<Title />).toJSON()

    expect(tree).toMatchStyledComponentsSnapshot()
  })

  test('call expression', () => {
    const Input = styled.input`
      color: attr(color);
      width: attr(width %);
      margin: attr(margin px, 16);
    `

    const H1 = styled('h1')`
      font-size: attr(fontSize);
      margin: attr(margin rem, 4);
    `

    const tree = renderer
      .create(
        <H1 className={'legacy__class'}>
          hello world
          <Input color="#ffffff" width={48} margin={16} />
        </H1>
      )
      .toJSON()

    expect(tree).toMatchStyledComponentsSnapshot()
  })

  test('function in expression', () => {
    const fontSize = 20
    const H1 = styled('h1')`
      height: attr(height);
      font-size: ${fontSize}px;
    `

    const H2 = styled(H1)`font-size: ${({ scale }) => fontSize * scale}`

    const tree = renderer
      .create(
        <H2 scale={2} height={48} className={'legacy__class'}>
          hello world
        </H2>
      )
      .toJSON()

    expect(tree).toMatchStyledComponentsSnapshot()
  })

  test('default value does not show when real value is 0', () => {
    const H1 = styled('h1')`
      height: attr(height px, 99);
    `

    const tree = renderer.create(<H1 height={0} />).toJSON()

    expect(tree).toMatchStyledComponentsSnapshot()
  })
})
