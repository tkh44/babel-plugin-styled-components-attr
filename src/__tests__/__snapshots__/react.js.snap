// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react attr 1`] = `
.hKjquV {
  font-size: 48;
  margin: 4rem;
  color: #495057;
}

<h1
  className="sc-bdVaJa hKjquV"
  color="#495057"
  fontSize={48}
/>
`;

exports[`react call expression 1`] = `
.etaRTp {
  color: #ffffff;
  width: 48%;
  margin: 16;
}

.jQCmMp {
  font-size: ;
  margin: 4rem;
}

<h1
  className="legacy__class sc-htpNat jQCmMp"
>
  hello world
  <input
    className="sc-bwzfXH etaRTp"
    color="#ffffff"
    width={48}
  />
</h1>
`;

exports[`react default value does not show when real value is 0 1`] = `
.hmEnUa {
  height: 99px;
}

<h1
  className="sc-EHOje hmEnUa"
  height={0}
/>
`;

exports[`react function in expression 1`] = `
.dFwslv {
  height: 48;
  font-size: 20px;
}

.gXhtwV {
  font-size: 40;
}

<h1
  className="legacy__class sc-ifAKCX gXhtwV sc-bxivhb dFwslv"
  height={48}
  scale={2}
>
  hello world
</h1>
`;
